wiki/middleware.py

```diff
@@ -0,0 +1,27 @@
+import threading
+
+_thread_locals = threading.local()
+
+
+def get_current_request():
+    """Return the request associated with the current thread."""
+    return getattr(_thread_locals, 'request', None)
```

Comment:
If you go that route you will have to add it conditionally. For example like: https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/django.py#L49-L52

But yes, I would prefer monkey-patching this method when loading `lms/djangoapps/course_wiki/` approach too.